From 603dda4779c13f28028ede6989585b02e30c559e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:28:07 -0400
Subject: [tools] perf/probe: Add '--demangle'/'--no-demangle'

Message-id: <1408462094-14194-319-git-send-email-jolsa@redhat.com>
Patchwork-id: 88286
O-Subject: [PATCH RHEL7.1 BZ1131394 318/325] perf probe: Add '--demangle'/'--no-demangle'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 35e17b2450e09968f9702d4048c228199af171bc
Author: Azat Khuzhin <a3at.mail@gmail.com>
Date: Mon Oct 28 12:04:24 2013 +0400

description
===========
You can't pass demangled name into "perf probe", because of special chars:
./perf probe -f -x /tmp/a.out 'foo(int)'
Semantic error :There is non-digit char in line number.

And you can't even pass without demangling (because it search symbol in
DSO with demangle=true):
./perf probe -f -x /tmp/a.out _Z3fooi
no symbols found in /tmp/a.out, maybe install a debug package?

However:
nm /tmp/a.out | grep foo
000000000040056d T _Z3fooi

After this patch, using the next command:
./perf probe -f --no-demangle -x /tmp/a.out _Z3fooi

probe will be successfully added.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-probe.c b/tools/perf/builtin-probe.c
index 89acc17..6ea9e85 100644
--- a/tools/perf/builtin-probe.c
+++ b/tools/perf/builtin-probe.c
@@ -325,6 +325,8 @@ int cmd_probe(int argc, const char **argv, const char *prefix __maybe_unused)
        opt_set_filter),
  OPT_CALLBACK('x', "exec", NULL, "executable|path",
    "target executable name or path", opt_set_target),
+ OPT_BOOLEAN(0, "demangle", &symbol_conf.demangle,
+      "Disable symbol demangling"),
  OPT_END()
  };
  int ret;
-- 
1.7.1